### PR TITLE
docs: Add full PR example to contributing guide

### LINKED ISSUE #66: Create a concrete PR creation example in the contributing guide
The contributing guide explains at a high level how to create a PR. Let's be a little more friendly and helpful and include a complete example.

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a full example of creating a PR including explaining how to rebase,
squash commits and force push to include review changes.

Fixes #66.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hopefully, this will be a gentle / non-scary quickstart for folk new to the dark arts of `git` magic...

ping @kata-containers/community.

**Comment by jodh-intel** [comment]
ping @kata-containers/documentation too.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -120,14 +122,237 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  This will create your copy under `https://github.com/${your-github-username}` *on the github server*.
+
+- [Clone your fork](https://help.github.com/articles/cloning-a-repository):
+
+  "Cloning" is a term that means to create a *local copy* of a git repository
+  that generally lives on a remote server.
+
+  ```sh
+  $ user="your-github-username"
+  $ dir="$GOPATH/src/github.com/kata-containers"
```
Comment (grahamwhaley) [inline CONTRIBUTING.md]: As you reference `GOPATH`, and this won't work if it is not set, you are probably going to have to briefly note that golang must be installed, and your GOPATH must be set up in the env.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -120,14 +122,237 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  $ mkdir -p "$dir"
+  $ cd "$dir"
+  $ git clone https://github.com/${user}/community
```
Comment (grahamwhaley) [inline CONTRIBUTING.md]: and maybe if you are using `GOPATH`, you could use `go get` - sure, I understand the quandry here, and a preference to probably stick with vanilla `git`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -120,14 +122,237 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  - The original "upstream" one (https://github.com/kata-containers/community)
+
+    You do not (and cannot) make changes to this repository. PRs are a way of
+    allowing the project developers and administrators to review your work and
+    decide if they want to add your work to the upstream branch.
+
+    > **Note:** Don't worry - even if you make a mistake,
+    > you *cannot* do any damage to this repository.
+
+  - Your fork (`https://github.com/${your-github-username}`) on GitHub
```
Comment (grahamwhaley) [inline CONTRIBUTING.md]: `${your-github-username}/community` ?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -120,14 +122,237 @@ rework your branch. You should rework any relevant information from the GitHub
...
+    repository.
+
+- [Setup a "git remote"](https://help.github.com/articles/configuring-a-remote-for-a-fork/):
+
+  You now have a local copy of your fork. However, you need to tell `git` where
+  your fork lives. The command below simply creates an alias (name) for the URL
+  of the fork you created above. This makes dealing with forks easier.
+
+  ```sh
+  $ git remote add github https://github.com/${user}/community
```
Comment (grahamwhaley) [inline CONTRIBUTING.md]: Will the git clone not have set up the `origin` remote already? If this is different, we may want to note how/why here (such as ssh vs http access)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -120,14 +122,237 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  ```
+
+- Create a new branch to do your work on:
```
Comment (grahamwhaley) [inline CONTRIBUTING.md]: ...new branch (known as a topic branch) to do...

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -120,14 +122,237 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  > **Note:** The `-f` option is the magic "force push". Since you have
+  > created a new commit using `git rebase`, you need to "overwrite" the old
+  > copy of your branch in your fork on GitHub. This might sound drastic but
+  > its similar to saving a file in your editor - you are simply overwriting
+  > the old version that you no longer want. Force pushing is the GitHub
+  > recommended way to handle an update to your PR - it also means you do not
+  > have to create a new PR on GitHub; force pushing will effectively re-use
+  > the initial PR you raised, but will include your latest changes.
+
+  For more information on GitHub "force push" workflows see "[Why and how to
```
Comment (grahamwhaley) [inline CONTRIBUTING.md]: Did you intend to indent this paragraph, but not make it a '>' comment type - just noting that is unusual in our project doc style.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -120,14 +122,361 @@ rework your branch. You should rework any relevant information from the GitHub
...
+
+  The way you keep your local copy up to date is to run `git pull`
+  periodically. This command will download all the latest changes from "another
+  repository" into your local repository. So how does `git` know which
+  repository to pull changes from? In simple terms it will default to pulling
+  from the repository you `git clone`'d above. Using `git` terminology, that
+  default repository is in fact given *two* names:
+
+  - The URL for the default upstream repository is called the `origin`.
+  - The for the default upstream repository is called the `origin`.
```
Comment (nitkon) [inline CONTRIBUTING.md]: Nit: "The fork of the default upstream repository"
Btw, nice explanation. If you would have written a book on it, I would have bought it. ;-)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -120,14 +122,361 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  commit. Assuming you followed the [example first PR](#example-first-pr),
+  your first commit ("`2e335ac docs: Fix typos and fold long lines`") will
+  already be in the [correct format](#patch-format). Therefore all you need to
+  do is keep the text for your first commit and delete everything else in the
+  editor window, updating it if appropriate based on the review feedback.
+
+  Save the file and quit the editor. Once this operation completes, the four
+  commits will have been converted into a single new commit. You can check this
+  by running the `git log command again:
+
```
Comment (nitkon) [inline CONTRIBUTING.md]: Nit: `git log`

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -120,14 +122,361 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  >   version that you no longer want. Force pushing is the GitHub
+  >   recommended way to handle an update to your PR - it also means you do
+  >   not have to create a new PR on GitHub; force pushing will effectively
+  >   re-use the initial PR you raised, but will include your latest changes.
+  >
+  >   For more information on GitHub "force push" workflows see "[Why and how to
+  >   correctly amend GitHub pull
+  >   requests](http://blog.adamspiers.org/2015/03/24/why-and-how-to-correctly-amend-github-pull-requests/)".
+
+Your PR will now have been updated on GitHub. To ensure team members are aware
```
Comment (nitkon) [inline CONTRIBUTING.md]: Nit: Your PR is now updated on GitHub?

**Comment by jodh-intel** [comment]
Thanks @nitkon - branch updated! ;)

**Comment by nitkon** [comment]
lgtm.
ping @kata-containers/community

**Comment by jodh-intel** [comment]
ping @klynnrif, @kata-containers/community

File: CONTRIBUTING.md (hunk 4 above)
Comment (marcov) [reply CONTRIBUTING.md]: I suggest to name the remote `fork`, to make it consistent with what's written above.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -120,14 +122,363 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  >   version that you no longer want. Force pushing is the GitHub
+  >   recommended way to handle an update to your PR - it also means you do
+  >   not have to create a new PR on GitHub; force pushing will effectively
+  >   re-use the initial PR you raised, but will include your latest changes.
+  >
+  >   For more information on GitHub "force push" workflows see "[Why and how to
+  >   correctly amend GitHub pull
+  >   requests](http://blog.adamspiers.org/2015/03/24/why-and-how-to-correctly-amend-github-pull-requests/)".
+
+Your PR is now updated on GitHub. To ensure team members are aware of this, it
```
Comment (klynnrif) [inline CONTRIBUTING.md]: Lines 477-480 suggested rewrite: 
... To ensure team member are aware of this, leave a message on the PR stating something like, "review feedback applied". Then, the team is notified and able to re-review your PR more quickly.

**Comment by jodh-intel** [comment]
Thanks for reviewing @marcov and @klynnrif - branch updated.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -120,14 +122,363 @@ rework your branch. You should rework any relevant information from the GitHub
...
+
+> **Note:** This section assumes no previous `git(1)` setup or knowledge.
+
+- Create a free GitHub account
+
+  To raise PRs and create issues, you must have a GitHub account.
+
+- Basic Git setup
+
+  First, install the `git` package for your distribution. Second tell Git
```
Comment (marcov) [inline CONTRIBUTING.md]: Needs a comma after `Second` or some rewording.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -120,14 +122,363 @@ rework your branch. You should rework any relevant information from the GitHub
...
+  You need to tell Git that your changes should be uploaded to your fork on
+  the GitHub server when you `git push` This is done by adding what is called
+  a "remote":
+
+  ```sh
+  $ user="your-github-username"
+  $ git remote add github https://github.com/${user}/community
+  ```
+
+  > **Note:** The previous command adds an alias (name) for the URL of the
```
Comment (marcov) [inline CONTRIBUTING.md]: Truncated sentence needs to be completed.

File: CONTRIBUTING.md (hunk 11 above)
Comment (jodh-intel) [reply CONTRIBUTING.md]: Fixed.

File: CONTRIBUTING.md (hunk 12 above)
Comment (jodh-intel) [reply CONTRIBUTING.md]: Fixed.

**Comment by jodh-intel** [comment]
Ping @klynnrif et al.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -120,14 +122,364 @@ rework your branch. You should rework any relevant information from the GitHub
...
+
+- [Setup a "git remote"](https://help.github.com/articles/configuring-a-remote-for-a-fork):
+
+  "What happened to my fork?". The following demonstrates where your fork is
+  and how you use it.
+
+  Recall that your fork on the GitHub server is an *exact copy* of the
+  upstream repository that also lives on the GitHub server. That might not be
+  true moving forward because PRs are constantly merged into the upstream
+  repository. This means your fork will be out of date because is will not
```
Comment (klynnrif) [inline CONTRIBUTING.md]: is / it 
This means your fork will be out of date because it will not

Comment (jodh-intel) [reply CONTRIBUTING.md]: Good catch @klynnrif - fixed!

**Comment by jodh-intel** [comment]
Ping @kata-containers/community - one more ack needed over here...

**Pull request merged**